modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -778,4 +778,26 @@ public void setPerformBeanValidation(boolean performBeanValidation) {
     public void setUseOptional(boolean useOptional) {
         this.useOptional = useOptional;
     }
+
+
+    @Override
+    public CodegenParameter fromParameter(Parameter parameter, Set<String> imports) {
+        CodegenParameter codegenParameter = super.fromParameter(parameter, imports);
+        Schema parameterSchema = parameter.getSchema();
+        if (parameterSchema != null) {
+            codegenParameter.defaultValue = getSpecifiedDefaultValue(parameterSchema);
+        }
+        return codegenParameter;
+    }
+
+    public String getSpecifiedDefaultValue(Schema p) {
+        if (ModelUtils.isArraySchema(p)) {
+            return null;
+        } else if (ModelUtils.isMapSchema(p)) {
+            return null;
+        }
+
+        return super.toDefaultValue(p);
+    }
+
```

Comment:
@rubms thanks for the fix. What about overriding `toDefaultValue` instead?

Ref: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L2816

Reply:
That's what I first tried, but it breaks the generated code: the default initialization of variables gets broken, causing the generated code not to compile. For instance, overriding `toDefaultValue`, if no default value is specified in the spec for an array it will be initialized to nothing (instead of new ArrayList<>()), generating a non-compiling code:
```java
List<String> listProperty = ;
```

This is why I overrode only the behavior for operation parameters, and not `toDefaultValue`.

Reply:
Thanks for the explanation. I think I know what issue you encountered.

I just fix a similar issue in another Java-related generator. Let me file a PR and have you to review my approach before deciding on which one to use.

Reply:
I've filed https://github.com/OpenAPITools/openapi-generator/pull/1725. Please review when you've time.

Moving forward, we may need to introduce toParameterDefaultValue and toPropertyDefaultValue for better control of default value in different scenarios.

Reply:
Sorry it took so long, I was on holidays. I agree, we need to introduce toParameterDefaultValue and toPropertyDefaultValue  to better control both scenarios. I have reviewed the #1725 PR and it looks good to me!